symfit/core/fit.py

```diff
@@ -465,6 +467,14 @@ def numerical_components(self):
              '`numerical_components`.').format(self.__class__)
         )
 
+    def _get_params(self):
+        return self._params
+
+    def _set_params(self, value):
+        self._params = value
+        self.__signature__ = self._make_signature()
+    params = property(_get_params, _set_params)
```

Comment:
Why not use the `@property` decorator?

Reply:
Unfortunately you cannot call super from within a `property` :(. This means that a construction like the one above is the only way to avoid code reuse. It looks dreadful internally I know, but at least from the outside API this is equivalent.

Reply:
Add a comment about this